src/codegen_envoy.rs

```diff
@@ -370,69 +184,212 @@ impl CodeGen for CodeGenEnvoy {
             .node_weight(trace_node_idx)
             .unwrap()
             .1
-            .contains_key(\"{prop}\")
+            .contains_key(\"{property}\")
         {{
             // we have not yet collected the return property
-            log::error!(\"Missing return property {prop}\");
+            log::error!(\"Missing return property {property}\");
             return None;
         }}
-        let ret_{prop_var} = &stored_data.trace_graph.node_weight(trace_node_idx).unwrap().1[ \"{prop}\" ];\n
-        value = ret_{prop_var}.to_string();\n",
+        let ret = &stored_data.trace_graph.node_weight(trace_node_idx).unwrap().1[ \"{property}\" ];\n
+        value = ret.to_string();\n",
                 node_id = entity,
-                prop_var = prop_wo_periods,
-                prop = property
+                property = property
         );
 
-        self.response_blocks.push(ret_block);
-    }
+    ret_block
+}
 
-    fn make_return_block(&mut self) {
-        if self.ir.return_expr.is_none() {
-            return;
-        }
-        let entity = self.ir.return_expr.as_ref().unwrap().clone().entity;
-        let mut property = self.ir.return_expr.as_ref().unwrap().clone().property;
-        if property.chars().next().unwrap() == ".".chars().next().unwrap() {
-            property.remove(0);
+fn make_return_block(entity_ref: &PropertyOrUDF, query_data: &VisitorResults) -> String {
+    match entity_ref {
+        PropertyOrUDF::Property(prop) => match prop.parent.as_str() {
+            "trace" => make_storage_rpc_value_from_trace(query_data.root_id.clone(), &prop.parent),
+            _ => make_storage_rpc_value_from_target(&prop.parent, &prop.to_dot_string()),
+        },
+        PropertyOrUDF::UdfCall(call) => {
+            // Because of quirky design we need to get the first arg
+            if call.args.len() != 1 {
+                panic!("We currently only implement very specific arguments for UDFs!");
+            }
+            let node = &call.args[0];
+            match node.as_str() {
+                "trace" => make_storage_rpc_value_from_trace(query_data.root_id.clone(), &node),
+                _ => make_storage_rpc_value_from_target(&node, &call.to_ref_str()),
+            }
         }
+    }
+}
 
-        if entity == "trace" {
-            self.make_storage_rpc_value_from_trace(self.ir.root_id.clone(), property);
-        } else {
-            let num_struct_filters = self.ir.struct_filters.len();
-            if !self.ir.struct_filters[num_struct_filters - 1]
-                .vertices
-                .contains(&entity)
-            {
-                panic!("Unknown entity in return expression");
-            }
-            self.make_storage_rpc_value_from_target(entity, property);
+fn make_aggr_block(agg: &Aggregate, query_data: &VisitorResults) -> String {
+    make_return_block(&agg.property, query_data)
+}
```

Comment:
Is there any functional difference between having this and just the return block?  Would it be possible to get rid of this function altogether?

Reply:
I kept it because it seems like it should serve some purpose. Currently, we only use the second part of the aggregation expression, making the aggregation effectively useless.

Reply:
Makes sense.